### ISSUE TITLE
sqlite: RuntimeError: Grammar refers to 'RespectKeywordSegment'

### ISSUE THREAD

**Issue by TommasoAmici** [sqlfluff/sqlfluff#4600]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I'm running sqlfluff lint (I tried both the CLI and the vscode extension) on a file with a single query and I get the following error:

```
RuntimeError: Grammar refers to 'RespectKeywordSegment' which was not found in the sqlite dialect.
```

As far as I can tell I'm using pretty run of the mill keywords in my query.

### Expected Behaviour

No error

### Observed Behaviour

<details>
<summary>Stack trace</summary>

<pre>
WARNING    Unable to lint time_per_language.sql due to an internal error. Please report this as an issue with your query's contents and stacktrace below!                                                                                                                                                                
To hide this warning, add the failing file to .sqlfluffignore
Traceback (most recent call last):
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/linter/runner.py", line 108, in run
    yield partial()
          ^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/linter/linter.py", line 787, in lint_rendered
    parsed = cls.parse_rendered(rendered)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/linter/linter.py", line 396, in parse_rendered
    parsed, pvs = cls._parse_tokens(
                  ^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/linter/linter.py", line 226, in _parse_tokens
    parsed: Optional[BaseSegment] = parser.parse(
                                    ^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/parser.py", line 39, in parse
    parsed = root_segment.parse(parse_context=ctx)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/base.py", line 1287, in parse
    self.segments = self.expand(
                    ^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/base.py", line 574, in expand
    res = stmt.parse(parse_context=parse_context)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/base.py", line 1287, in parse
    self.segments = self.expand(
                    ^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/base.py", line 574, in expand
    res = stmt.parse(parse_context=parse_context)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/base.py", line 1287, in parse
    self.segments = self.expand(
                    ^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/base.py", line 574, in expand
    res = stmt.parse(parse_context=parse_context)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/base.py", line 1287, in parse
    self.segments = self.expand(
                    ^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/base.py", line 574, in expand
    res = stmt.parse(parse_context=parse_context)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/base.py", line 1287, in parse
    self.segments = self.expand(
                    ^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/base.py", line 574, in expand
    res = stmt.parse(parse_context=parse_context)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 51, in parse
    new_self = super().parse(parse_context, parse_grammar=self._parse_grammar)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/base.py", line 1287, in parse
    self.segments = self.expand(
                    ^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/base.py", line 574, in expand
    res = stmt.parse(parse_context=parse_context)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/base.py", line 1287, in parse
    self.segments = self.expand(
                    ^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/base.py", line 574, in expand
    res = stmt.parse(parse_context=parse_context)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/base.py", line 1223, in parse
    m = parse_grammar.match(segments=segments, parse_context=ctx)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 103, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/base.py", line 942, in match
    resp = elem.match(segments=segments, parse_context=ctx)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 103, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/sequence.py", line 143, in match
    elem_match = elem.match(mid_seg, parse_context=ctx)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 103, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/delimited.py", line 154, in match
    match, _ = self._longest_trimmed_match(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/base.py", line 262, in _longest_trimmed_match
    res_match = matcher.match(segments, parse_context=parse_context)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 103, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/base.py", line 942, in match
    resp = elem.match(segments=segments, parse_context=ctx)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/base.py", line 798, in match
    m = cls.match_grammar.match(segments=segments, parse_context=ctx)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 103, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 216, in match
    match, matched_option = self._match_once(
                            ^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 152, in _match_once
    match, matched_option = self._longest_trimmed_match(
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/base.py", line 262, in _longest_trimmed_match
    res_match = matcher.match(segments, parse_context=parse_context)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 103, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/sequence.py", line 143, in match
    elem_match = elem.match(mid_seg, parse_context=ctx)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 103, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/base.py", line 942, in match
    resp = elem.match(segments=segments, parse_context=ctx)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 103, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 216, in match
    match, matched_option = self._match_once(
                            ^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 152, in _match_once
    match, matched_option = self._longest_trimmed_match(
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/base.py", line 262, in _longest_trimmed_match
    res_match = matcher.match(segments, parse_context=parse_context)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 103, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/base.py", line 942, in match
    resp = elem.match(segments=segments, parse_context=ctx)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/base.py", line 798, in match
    m = cls.match_grammar.match(segments=segments, parse_context=ctx)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 103, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 216, in match
    match, matched_option = self._match_once(
                            ^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 152, in _match_once
    match, matched_option = self._longest_trimmed_match(
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/base.py", line 262, in _longest_trimmed_match
    res_match = matcher.match(segments, parse_context=parse_context)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 103, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/sequence.py", line 143, in match
    elem_match = elem.match(mid_seg, parse_context=ctx)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 103, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/base.py", line 942, in match
    resp = elem.match(segments=segments, parse_context=ctx)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 103, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 185, in match
    available_options, _ = self._prune_options(
                           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 82, in _prune_options
    simple = opt.simple(parse_context=parse_context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/base.py", line 72, in wrapped_method
    result = func(self, parse_context=parse_context, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/base.py", line 861, in simple
    return self._get_elem(dialect=parse_context.dialect).simple(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/segments/base.py", line 701, in simple
    return cls.match_grammar.simple(parse_context=parse_context, crumbs=crumbs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/base.py", line 72, in wrapped_method
    result = func(self, parse_context=parse_context, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/sequence.py", line 41, in simple
    simple = opt.simple(parse_context=parse_context, crumbs=crumbs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/base.py", line 72, in wrapped_method
    result = func(self, parse_context=parse_context, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/sequence.py", line 41, in simple
    simple = opt.simple(parse_context=parse_context, crumbs=crumbs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/base.py", line 72, in wrapped_method
    result = func(self, parse_context=parse_context, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 42, in simple
    simple_buff = [
                  ^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 43, in <listcomp>
    opt.simple(parse_context=parse_context, crumbs=crumbs)
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/base.py", line 72, in wrapped_method
    result = func(self, parse_context=parse_context, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/base.py", line 861, in simple
    return self._get_elem(dialect=parse_context.dialect).simple(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/parser/grammar/base.py", line 889, in _get_elem
    return dialect.ref(self._get_ref())
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tommi/.local/lib/python3.11/site-packages/sqlfluff/core/dialects/base.py", line 278, in ref
    raise RuntimeError(
RuntimeError: Grammar refers to 'RespectKeywordSegment' which was not found in the sqlite dialect. Perhaps specify the keyword? https://github.com/sqlfluff/sqlfluff/wiki/Contributing-Dialect-Changes#keywords
 
All Finished 📜 🎉!
```
</pre>

</details>

### How to reproduce

With the query below, run the following command:

```sh
sqlfluff lint --dialect sqlite time_per_language.sql
```

```sql
WITH time_cte AS (
  SELECT
      branch,
      created_at,
      time,
      cast(time - LAG (time, 1, time) OVER (ORDER BY time) as real) AS time_spent
  FROM heartbeats h
  WHERE user_id = ? AND created_at >= DATE('now', 'start of day')
  ORDER BY id
  LIMIT -1 OFFSET 1
)
SELECT
    branch as name,
    cast(time_spent as real) as time_spent,
    cast(time_spent / (SELECT SUM(time_spent) FROM time_cte) as real) as time_percentage
FROM (
    SELECT
        branch,
        cast(SUM(time_spent) as real) AS time_spent
    FROM time_cte
    GROUP BY branch
    ORDER BY time_spent DESC
)
```

### Dialect

sqlite

### Version

sqlfluff, version 2.0.2
Python 3.11.2
macos 13.2.1

### Configuration

I don't have a sqlfluff configuration

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by greg-finley** [comment]
@WittierDinosaur thoughts on just adding non-reserved words even if they aren't?

i.e. this would get you past the error if you override ANSI to remove `RESPECT`:

```python
class OverClauseSegment(ansi.OverClauseSegment):
    """An OVER clause for window functions."""

    match_grammar: Matchable = Sequence(
        "OVER",
        OneOf(
            Ref("SingleIdentifierGrammar"),  # Window name
            Bracketed(
                Ref("WindowSpecificationSegment", optional=True),
            ),
        ),
    )
```

But then you hit: `RuntimeError: Grammar refers to 'QualifyKeywordSegment' which was not found in the sqlite dialect. Perhaps specify the keyword? https://github.com/sqlfluff/sqlfluff/wiki/Contributing-Dialect-Changes#keywords`

**Comment by WittierDinosaur** [comment]
I'm gonna take a look at this at the weekend. I think the ultimate best thing may be to not inherit from ANSI for SQLite. SQLite is so small, and generally unchanging, that it might be easier to keep it as a self-contained dialect

**Comment by tunetheweb** [comment]
Yeah this generally happens when a bit of SQL is unparsable so it ends up going down paths that it was never expected to go down for this dialect and then gives that very unhelpful error message.

The real problem is this:

```
==== parsing violations ====
L:   8 | P:  16 |  PRS | Line 8, Position 16: Found unparsable section: " = ? AND
                       | created_at >= DATE('now', 'star..."
L:   9 | P:  14 |  PRS | Line 9, Position 14: Found unparsable section: '\n  LIMIT -1
                       | OFFSET 1'
```

So currently we don't support either of those lines.

Adding support for them will remove this error, but we'll just get that for the next unparsable query.

So the options to prevent this nasty error message are:
1. Ensure the paths with keywords that aren't keywords can never be reached in this dialect (e.g. by overriding the classes that refer to them
2. Add all keywords referred to in the ANSI dialect as unreserved keywords. Not technically correct but often easier than above.
3. Stop inheriting from ANSI.
4. Change SQLFluff to handle these errors better (or ignore them?). Clearly it knows it's a missing keyword so why not just pretend it's defined and carry on, and then give the real parsing error later? Instead of showing this horrible, and incorrect, and useless error message.

I've used a 1. in some cases, and 2. in others. Never gone as far as 3. before. And would love for us to implement 4 as the fuller long term solution to this!

@alanmcruickshank WDYT of suggestion 4?
